src/utils/fileUpload.js

```diff
@@ -38,38 +76,74 @@ const getFileExtension = function(path) {
  * @param {string} userRoot user root path
  * @param {string} path The path whose existence in the destination is to
  * be checked
- * @return {string} The unique path
+ * @param {number} knownSuffix The suffix to start looking from
+ * @return {object} The unique path and suffix
  */
-const findUniquePath = async function(client, userRoot, path) {
+const findUniquePath = async function(client, userRoot, path, knownSuffix) {
 	// Return the input path if it doesn't exist in the destination folder
-	if (await client.exists(userRoot + path) === false) {
-		return path
+	if (!knownSuffix && await client.exists(userRoot + path) === false) {
+		return { uniquePath: path, suffix: 1 }
```

Comment:
For example, if test (6).txt path doesn't exist and there is no known Suffix yet, it will be stored as test (1).txt whereas its suffix should be 6. Either change the path or update suffix with the file's suffix.
As a default process, I am not sure if you want to force a suffix order ( first file is 1, second is 2 etc) regardless of their original suffixes or you want to store them with their original suffixes.  

[ Just a thought] If you are considering the first option, maybe there is a client function to get the list of all filenames that match fileName + ` (suffix = whatever in between)` + fileExtension and get its maximum ? In order to avoid looping.

Reply:
No, there won't be a file with (1) at all. If we try to upload 4 files with name `file (4).txt`:
- `knownSuffix === undefined`, `client.exists === false` => return `{'file (4).txt', 1}`
- `knownSuffix === 1` => return `{'file (2).txt', 2}`
- `knownSuffix === 2` => return `{'file (3).txt', 3}`
- `knownSuffix === 3` => return `{'file (5).txt', 5}` (cause 4 existing already).

Although I may agree if it keeps following from the original suffix: 4,5,6,7

There is a method to search via webdav client: https://www.npmjs.com/package/webdav#search, but
- `client.search()` is more expensive than `client.exists()`
- it returns `501 (Not Implemented)`, so doesn't work for us =)

Reply:
> .. is more expensive than client.exists()

Yeah, that's what I was skeptical about

>  Although I may agree if it keeps following from the original suffix: 4,5,6,7

I agree too as there is no use of a forced ordering.